R/deps.R

```diff
@@ -6,11 +6,40 @@
 #' @param src source that the package is installed from (CRAN or github)
 #' @param version version of the package
 #' @param install whether or not the package should be installed in the model image
-add.dependency <- function(name, importName, src, version, install) {
-  # Don't add the dependency if it's already there
+#' @param auth_token a personal access token for github or gitlab repositories
+#' @param ref The git branch, tag, or SHA of the package to be installed
```

Comment:
Since `ref` is optional (it is optional, right?) should we add a note in here that it will default to the master branch and latest commit if left out?

Reply:
None of the parameters are optional in `add.dependency`. This function is only called internally by `promote.library`, which gets called by the user. And in `promote.library`,  some of these are optional, but in order to always have the right columns in the dataframe, add.dependency will get a value for all (even if NULL, which is then converted to NA).